apps/common/lib/lexical/ast/detection/alias.ex

```diff
@@ -23,37 +23,42 @@ defmodule Lexical.Ast.Detection.Alias do
   """
   @impl Detection
   def detected?(%Analysis{} = analysis, %Position{} = position) do
-    analysis.document
-    |> Tokens.prefix_stream(position)
-    |> Stream.with_index()
-    |> Enum.to_list()
-    |> Enum.reduce_while(false, fn
-      {{:curly, :"{", _}, _index}, :eol ->
-        {:cont, false}
+    result =
+      analysis.document
+      |> Tokens.prefix_stream(position)
+      |> Stream.with_index()
+      |> Enum.reduce_while(false, fn
+        {{:curly, :"{", _}, _index}, :eol ->
+          {:cont, false}
 
-      {{:comma, _, _}, _index}, :eol ->
-        {:cont, false}
+        {{:comma, _, _}, _index}, :eol ->
+          {:cont, false}
 
-      {{:eol, _, _}, _index}, _acc ->
-        {:cont, :eol}
+        {{:eol, _, _}, _index}, _acc ->
+          {:cont, :eol}
 
-      {{_, _, _}, _}, :eol ->
-        {:halt, false}
+        {{_, _, _}, _}, :eol ->
+          {:halt, false}
 
-      {{:curly, :"}", _}, _index}, _ ->
-        {:halt, false}
+        {{:curly, :"}", _}, _index}, _ ->
+          {:halt, false}
 
-      {{:identifier, ~c"alias", _}, 0}, _ ->
-        # there is nothing after the alias directive, so we're not
-        # inside the context *yet*
+        {{:identifier, ~c"alias", _}, 0}, _ ->
+          # there is nothing after the alias directive, so we're not
+          # inside the context *yet*
 
-        {:halt, false}
+          {:halt, false}
 
-      {{:identifier, ~c"alias", _}, _index}, _ ->
-        {:halt, true}
+        {{:identifier, ~c"alias", _}, _index}, _ ->
+          {:halt, true}
 
-      _, _ ->
-        {:cont, false}
-    end)
+        _, _ ->
+          {:cont, false}
+      end)
+
+    case result do
+      b when is_boolean(b) -> b
+      _ -> false
+    end
```

Comment:
Being unfamiliar with this code, it took a while for me to figure out why this was necessary. I think being explicit about the non-boolean value that we might expect would make this more clear.

```suggestion
    case result do
      b when is_boolean(b) -> b
      :eol -> false
    end
```

Reply:
yes, but `fetch_range` takes a document, which makes the api a lot more cumbersome.
Can you provide an example of sourceror returning nil in these cases? I know it's possible, but I was under the impression that it will only do so if there is no information in the AST, which is not the case for our usages.